### DISCUSSION TITLE
Exclude files from being counted under code frequency

### DISCUSSION THREAD

**Discussion by engelsen** [community/community#21858] [New to GitHub]
<p>Hi all,</p>
<p>I am working on a project in MATLAB to do some basic data processing and instrument communication, for which I have made a GitHub repository (<a href="https://github.com/engelsen/Instrument-control" rel="nofollow noopener">https://github.com/engelsen/Instrument-control</a>). Recently, I committed a couple of data files (tab delimited .txt files) for a test case. These are now being counted as code under code frequency, despite my attempt to tell linguist that I would like it to ignore these files. Can anyone see what is going wrong here? I made a .gitattributes file that only contains the following:</p>
<pre><code>*.txt linguist-documentation
</code></pre>
<p>Thanks for any help.</p>

**Comment by lee-dohm** [comment]
<p>It looks like your project is being reported as 100% MatLab at this point. The job that calculates code frequency doesn’t run immediately so it can take a little while for things to update once you’ve uploaded a .gitattributes file. Things look good now though :+1:</p>

**Comment by engelsen** [comment]
<p>Hi, thanks for the reply. I can see that it says 100% MATLAB, so it seems that part is indeed working well. However, under Insights-&gt;Pulse, it claims that I have so far made 21390 additions, which is apparently counting the data files I uploaded. Is there any way to fix this? </p>

**Comment by lee-dohm** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/engelsen">@engelsen</a> wrote:</p>
</blockquote>
<blockquote>
<p>it claims that I have so far made 21390 additions, which is apparently counting the data files I uploaded. Is there any way to fix this? </p>
</blockquote>
<p>No, there isn’t. If you want to exclude data files from all of the stats reported about your repository then you should probably store the data files somewhere outside of the repository.</p>

**Comment by dsokhach** [comment]
<p>The problem described by <a class="mention" href="/engelsen">@engelsen</a>  is not related to git repo, but to the statistics counting tool.</p>
<p>The tool is not flexible enough to support required scenario.</p>
<p>On my projects I also have “data” files or “generated code” which I must commit to the repo.</p>
<p>package-lock.json can take 40 000 lines and must be commited for repo to work properly.</p>
<p>Test data can take kilo-megabytes and also msut be commited for tests to work for everyone who clones the repo, including CI machines.</p>
<p>I wasn’t able to find “dislike” or “lower votes” for this message and think it is inappropriate solution.</p>

**Comment by bholbrook73** [comment]
<p><code>.package-lock.json</code> is a perfect example of a file you don’t want to count but does belong in the repo.</p>

**Reply by fohrloop** [reply]
Also `.svg` files for docs.

**Comment by jwickens** [comment]
Try this https://proandroiddev.com/removing-noise-from-your-github-language-stats-e96113f8183d

**Reply by basememara** [reply]
This is interesting but seems to only work locally not on the GitHub PR for example. I tried this in my `.gitattributes`:

```
my-generated-file.d.ts  diff=nodiff
```

It seemed to have no effect on the PR.

**Reply by jwickens** [reply]
Yes sadly @basememara Github PR diff counts do not respect this (as I found out after posting my message last year). But they should!

**Comment by qdm12** [comment]
Same thing. Data files sometimes need to be updated in the repository, but it feels silly it's counting as additions/deletions, and it would be nice to ignore specific files (preferrably retro-actively too, for example let the user set the start date and end date optionally for a stat ignored file path).

**Reply by Blue-Robin-Taken** [reply]
Definitely,
even though it's been a while since this post was up
I think that there should be something added to exclude files from additions/deletions because:
- It'd look cooler
- It'd be a better measurement of work

**Comment by d-beloved** [comment]
I found this issue a few days ago because I was also in this dilemma, I wanted to exclude some files from being counted as part of code_frequency, so I built this tool https://www.npmjs.com/package/git-repo-lines-of-code/v/1.0.2, this solved the problem for me. You will pass in the relative paths of the files you want to exclude as an array of strings along with your repository details also.
